src/test/perl/path.t

```diff
@@ -108,9 +108,9 @@ sub verify_exception
     } else {
         ok(! $ec_check->error(), "Error reset after $msg");
     };
-    if ($noreset) {
+    if ($noreset && $mc->{fail}) {
```

Comment:
@jouvin this still needs to be addressed. i would revert the changes and keep the old logic. it's in any case more important that it fails rather that the message is readable/human friendly